bin/node-template/node/src/command.rs

```diff
@@ -15,32 +15,35 @@
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
 use sp_consensus_aura::sr25519::{AuthorityPair as AuraPair};
-use sc_cli::{VersionInfo, error};
+use sc_cli::VersionInfo;
 use crate::service;
 use crate::chain_spec;
 use crate::cli::Cli;
 
 /// Parse and run command line arguments
-pub fn run(version: VersionInfo) -> error::Result<()> {
+pub fn run(version: VersionInfo) -> sc_cli::Result<()> {
 	let opt = sc_cli::from_args::<Cli>(&version);
 
-	let config = sc_service::Configuration::new(&version);
+	let mut config = sc_service::Configuration::from_version(&version);
 
 	match opt.subcommand {
-		Some(subcommand) => sc_cli::run_subcommand(
-			config,
-			subcommand,
-			chain_spec::load_spec,
-			|config: _| Ok(new_full_start!(config).0),
-			&version,
-		),
-		None => sc_cli::run(
-			config,
-			opt.run,
-			service::new_light,
-			service::new_full,
-			chain_spec::load_spec,
-			&version,
-		)
+		Some(subcommand) => {
+			subcommand.init(&version)?;
```

Comment:
Why aren't `init` and `update_config` not done inside of `run()`? Someone could just call `run()` without calling the other methods before and it would be incorrect.

Reply:
Because for cumulus we want to be able to modify the config before running the node.

We could have another "run" command that would do all 3 commands but we need to keep the existing 3.

Since a helper would simply call init & update_config & run, I thought it wouldn't be super helpful and I have not added it.

Reply:
> I removed the run and run_subcommand helpers as they are not helping much anymore. Running a command is always a set of 3 commands: 1. init 2. update config 3. run. This still allow the user to change the config before arguments get parsed or right after.

:scream: you didn't read the PR description

Reply:
Code is law :P 

Yeah it is okay for now. However it could happen that the user just calls `run()` (because nothing prevents them) and it will break.

Reply:
And this "pattern" is a let's repeat some code "pattern" :D